cli/src/commands/test.ts

```diff
@@ -18,35 +20,113 @@ export const handler = async (argv: Arguments<Options>): Promise<void> => {
-  await executeCommand({
-    command: "/bin/bash",
-    args: [
-      commandPath,
-      systemProgramPath,
-      process.cwd(),
-      programAddress,
-      `${programName}.so`,
-      `yarn ts-mocha -t 2000000 tests/${projectName}.ts --exit`
-    ]
-  });
+  await start_test_validator(projectName, programName, programAddress);
   process.exit(0);
 };
 
+export async function start_test_validator(
+  projectName: string,
+  programName: string,
+  programAddress: string
+) {
+  const command = "solana-test-validator";
+  const LIMIT_LEDGER_SIZE = "500000000";
+  const BASE_PATH = "/bin/";
+  const programs = [
+    { id: "noopb9bkMVfRPU8AsbpTUg8AQkHtKwMYZiFUjNRtMmV", name: "spl_noop.so" },
+    {
+      id: "JA5cjkRJ1euVi9xLWsCJVzsRzEkT8vcC4rqw9sVAo5d6",
+      name: "merkle_tree_program.so",
+    },
+    {
+      id: "J1RRetZ4ujphU75LP8RadjXMf3sA12yC2R44CF7PmU7i",
+      name: "verifier_program_zero.so",
+    },
+    {
+      id: "DJpbogMSrK94E1zvvJydtkqoE4sknuzmMRoutd6B7TKj",
+      name: "verifier_program_storage.so",
+    },
+    {
+      id: "J85SuNBBsba7FQS66BiBCQjiQrQTif7v249zL2ffmRZc",
+      name: "verifier_program_one.so",
+    },
+    {
+      id: "2cxC8e8uNYLcymH6RTGuJs3N8fXGkwmMpw45pY65Ay86",
+      name: "verifier_program_two.so",
+    },
+    { id: programAddress, path: `./target/deploy/${programName}.so` },
+  ];
+
+  const dirPath = path.resolve(__dirname, BASE_PATH);
+
+  let solanaArgs = [
+    "--reset",
+    `--limit-ledger-size=${LIMIT_LEDGER_SIZE}`,
+    "--quiet",
+  ];
+
+  for (let program of programs) {
+    let dirPathString = "../../bin/" + program.name;
+    const localFilePath = path.resolve(__dirname, dirPathString);
+    if (!program.path) {
+      await downloadBinIfNotExists({
+        localFilePath,
+        dirPath,
+        owner: "Lightprotocol",
+        repoName: "light-protocol",
+        remoteFileName: program.name,
+      });
+    }
+
+    let path1 = program.path ? program.path : `${localFilePath}`;
+    solanaArgs.push("--bpf-program", program.id, path1);
+  }
+  let dirPathString = "../../accounts/";
+  const localFilePath = path.resolve(__dirname, dirPathString);
+  console.log("accounts path ", localFilePath);
+  solanaArgs.push("--account-dir", localFilePath);
+
+  try {
+    // killall process
+    await executeCommand({
+      command: "killall",
+      args: ["solana-test-validator"],
+    });
+  } catch (error) {}
+  try {
+    // killall process
+    await executeCommand({
+      command: "killall",
+      args: ["solana-test-val"],
+    });
+  } catch (error) {}
+
+  await new Promise((r) => setTimeout(r, 1000));
+
+  executeCommand({
+    command,
+    args: [...solanaArgs],
+  });
+  await sleep(10000);
+  await executeCommand({
+    command: `yarn`,
+    args: [`ts-mocha`, `-t`, `2000000`, `tests/${projectName}.ts`, `--exit`],
+  });
+}
```

Comment:
I like this command a lot! But for the future, can we keep such change completely separate? Adding a CLI command is not "general cleanup" - it's a feature which deserves a separate PR, which doesn't touch anything else.

Reply:
I have no clue about that